src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -1192,9 +1192,12 @@ void StubGenerator::restore_arg_regs() {
 
 // This is used in places where r10 is a scratch register, and can
 // be adapted if r9 is needed also.
-void StubGenerator::setup_arg_regs_using_thread() {
+void StubGenerator::setup_arg_regs_using_thread(int nargs) {
   const Register saved_r15 = r9;
 #ifdef _WIN64
+  if (nargs >= 4) {
```

Comment:
When would `nargs` ever be `> 4`?

Reply:
It won't (yet). I was just structuring it like setup_arg_regs, where it similarly checks if nargs >= 4. But I can change both to an assert that it isn't > 4.

Reply:
An assert sounds good. This method (linux-part) gives me the impression that it's ensuring _four_ args are OK.